
Add the missing fontconfig.pc* files, support and code in the style of
Xcursor (I would do it cleaner - but it this is the XFree86-way ...)

  - Rene Rebe <rene@rocklinux.org>

--- xc/lib/fontconfig/Imakefile	2003-06-10 13:04:51.000000000 +0200
+++ xc-new/lib/fontconfig/Imakefile	2003-08-30 20:02:57.000000000 +0200
@@ -17,6 +17,12 @@
 
 HEADERS=fcfreetype.h fcprivate.h fontconfig.h
 
+SUBSTVARS=prefix="$(PROJECTROOT)" \
+          exec_prefix="$(BINDIR)" \
+          libdir="$(USRLIBDIR)" \
+          includedir="$(INCROOT)" \
+          PACKAGE_VERSION="1.0.1"
+
 BuildIncludes($(HEADERS),IncSubdir,..)
 InstallMultipleFlags($(HEADERS),$(INCDIR)/IncSubdir,$(INSTINCFLAGS))
 
@@ -91,12 +97,19 @@
 InstallNonExecFileBackup(fonts.conf,$(FONTCONFIGFONTSCONFDIR))
 InstallNonExecFile(fonts.dtd,$(FONTCONFIGFONTSCONFDIR))
 
-all:: fonts.conf
+all:: fonts.conf fontconfig.pc
 
 fonts.conf: fonts.conf.in setfontdirs findfonts
 	RemoveFile($@)
 	sh ./setfontdirs $(FONTDIRS)
 
+fontconfig.pc: fontconfig.pc.in
+	RemoveFile($@)
+	sh config-subst $(SUBSTVARS) < $@.in > $@
+
+InstallNonExecFile(xcursor.pc,$(USRLIBDIR)/pkgconfig)
+
+
 clean::
 	RemoveFile(fonts.conf)
 
--- xc/lib/fontconfig/fontconfig.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ xc-new/lib/fontconfig/fontconfig.pc.in	2003-08-30 19:09:46.000000000 +0200
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: Fontconfig
+Description: Font configuration and customization library
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -lfontconfig
+Cflags: -I${includedir}
--- xc/lib/fontconfig/config-subst	1970-01-01 01:00:00.000000000 +0100
+++ xc-new/lib/fontconfig/config-subst	2003-08-30 19:43:14.000000000 +0200
@@ -0,0 +1,11 @@
+#!/bin/sh
+set -x
+script=config-subst.$$
+trap "rm $script" 0
+rm -f $script
+for i in ${1+"$@"}; do
+	var="`echo "$i" | sed 's/=.*$//'`"
+	val="`echo "$i" | sed 's/^[^=]*=//'`"
+	echo "s;@$var@;$val;" >> $script
+done
+sed -f $script
